### ISSUE TITLE
Possible to use spack in the browser similar to esbuild-wasm

### ISSUE THREAD

**Issue by brynshanahan** [swc-project/swc#3203]
### Describe the feature

Would it be possible to use spack in the browser to bundle virtual files / URLs on the frontend? Ideally it would be similar to esbuild-wasm. 

I'd love to be able to move from esbuild to swc but I'd rather not have to maintain two separate bundling methods for server/browser

**Comment by kdy1** [comment]
I think it's possible, but I'm not sure how the esbuild-wasm works.
Fetching URL is asynchronous, right?

**Comment by brynshanahan** [comment]
Yeah esbuild only allows the async versions of the apis to be used

**Comment by marklundin** [comment]
@brynshanahan did you get any further with this? I’d be keen to find out if this is possible
